astropy/table/table.py

```diff
@@ -1182,8 +1182,8 @@ def more(self, max_lines=None, max_width=None, show_name=True,
     def __getitem__(self, item):
         if isinstance(item, six.string_types):
             return self.columns[item]
-        elif isinstance(item, (int, np.integer)):
-            return self.Row(self, item)
+        elif (isinstance(item, (int, np.integer)) or (isinstance(item, np.ndarray) and item.shape == () and item.dtype.kind == 'i')):
+            return self.Row(self, np.array(item).item())
```

Comment:
Is the `np.array(item).item()` necessary? If so, just split the two if-statements:
```
elif isinstance(item, (int, np.integer)):
    return self.Row(self, item)
elif isinstance(item, ndarray) and item.shape == () and item.dtype.kind == 'i':
    return self.Row(self, item.item())
```